src/mail-app/contacts/ContactImporter.ts

```diff
@@ -96,46 +103,71 @@ export class ContactImporter {
 	}
 
 	private async importContactsFromDevice() {
-		assert(isApp(), "isApp")
-		const contactBooks = await showProgressDialog("pleaseWait_msg", locator.mobileContactsFacade.getContactBooks())
-		let books: readonly ContactBook[]
-		if (contactBooks.length === 0) {
+		// these will only ever be null if !isApp()
+		const mobileContactsFacade = assertNotNull(this.mobileContactsFacade)
+		const nativeContactSyncManager = assertNotNull(this.nativeContactSyncManager)
+
+		const books = await this.selectContactBooks(mobileContactsFacade)
+		if (books == null) {
 			return
-		} else if (contactBooks.length === 1) {
-			books = contactBooks
-		} else {
-			const importDialog = new ImportNativeContactBooksDialog(contactBooks)
-			const selectedBooks = await importDialog.show()
-			if (selectedBooks == null || selectedBooks.length === 0) return
-			books = selectedBooks
 		}
 
 		const contactListId = await locator.contactModel.getContactListId()
 		const contactGroupId = await locator.contactModel.getContactGroupId()
-		const contactsToImport: Contact[] = (
+
+		const allImportableStructuredContacts: StructuredContact[] = (
 			await promiseMap(books, async (book) => {
-				const structuredContacts = await locator.mobileContactsFacade.getContactsInContactBook(
-					book.id,
-					locator.logins.getUserController().loginUsername,
-				)
-				return structuredContacts.map((contact) => this.contactFromStructuredContact(contactGroupId, contact))
+				return await mobileContactsFacade.getContactsInContactBook(book.id, locator.logins.getUserController().loginUsername)
 			})
 		).flat()
+		const allImportableContacts = allImportableStructuredContacts.map((contact) => this.contactFromStructuredContact(contactGroupId, contact))
 
 		const importer = await mailLocator.contactImporter()
-
 		showContactImportDialog(
-			contactsToImport,
-			(dialog, selectedContacts) => {
+			allImportableContacts,
+			async (dialog, selectedContacts) => {
 				dialog.close()
-				importer.importContacts(selectedContacts, assertNotNull(contactListId))
+				let selectedStructuredContacts: StructuredContact[]
+				if (selectedContacts.length === allImportableContacts.length) {
+					selectedStructuredContacts = allImportableStructuredContacts
+				} else {
+					selectedStructuredContacts = selectedContacts.map((contact) => allImportableStructuredContacts[allImportableContacts.indexOf(contact)])
+				}
+
+				await importer.importContacts(selectedContacts, assertNotNull(contactListId))
+				const imported = nativeContactSyncManager.isEnabled() && (await nativeContactSyncManager.syncContacts())
+
+				// On iOS, we want to give the option to remove the contacts locally, but we obviously only want to do
+				// this if syncing is successful, assuming syncing is enabled.
+				//
+				// Do nothing further if not on iOS, or if syncing is disabled or failed.
+				if (imported && isIOSApp()) {
```

Comment:
~~is this the only place where we try to deduplicate? This will probably not be enough for most cases (like onboarding) because import is done first and sync is enabled afterwards.~~

it is also done in `enableSync()`